### PR TITLE
sdk: CloudBackup Support

### PR DESCRIPTION
**Pull request by lpabon**
**What this PR does / why we need it**:
Provides CloudBackup calls to the SDK.

**Which issue(s) this PR fixes** (optional)
Part of #407

**Special notes for your reviewer**:
* Do not worry too much about the docs. There will be a full review of the docs next week.
* Take care reviewing the translations from SDK types to Golang interface types.
* Also, the generated files have been moved to a separate commit for convenience.

### CODE REVIEW

File: api/api.proto [hunk 1]
```diff
@@ -1373,3 +1457,185 @@ message SdkObjectstoreUpdateRequest {
...
+}
+
+message SdkCloudBackupDeleteResponse {
+}
+
+message SdkCloudBackupDeleteAllRequest {
+  // id of the volume for the request
+  string src_volume_id = 1;
+  // CredentialUUID is the credential for cloud to be used for the request
+  string credential_uuid = 3;
```
Comment (ram-infrac) [inline api/api.proto]: is there any specific reason to jump for parameter number here?

File: api/server/sdk/cloud_backup.go [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+	if err := s.driver.CloudBackupCreate(&api.CloudBackupCreateRequest{
+		VolumeID:       req.GetVolumeId(),
+		CredentialUUID: req.GetCredentialUuid(),
+		Full:           req.GetFull(),
+	}); err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to create backup: %v", err)
+	}
+
+	return &api.SdkCloudBackupCreateResponse{
+		// THIS WILL BE FIXED BY #### where Create will return an ID
```
Comment (ram-infrac) [inline api/server/sdk/cloud_backup.go]: Just curious about this, here should current ```CloudBackupCreate``` api should be modified to return backup ID right? Alternatively we can enumerate and match volume ID with newly created cloudbackup and send that back again

File: api/server/sdk/cloud_backup.go [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+
+// Restore a backup
+func (s *CloudBackupServer) Restore(
+	ctx context.Context,
+	req *api.SdkCloudBackupRestoreRequest,
+) (*api.SdkCloudBackupRestoreResponse, error) {
+
+	if len(req.GetBackupId()) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Must provide backup id")
+	} else if len(req.GetCredentialUuid()) == 0 {
```
Comment (ipochi) [inline api/server/sdk/cloud_backup.go]: nit: do we really need else statements 
couple of if's would do , idiomatic go.

File: api/server/sdk/cloud_backup.go [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+
+// Delete deletes a backup
+func (s *CloudBackupServer) Delete(
+	ctx context.Context,
+	req *api.SdkCloudBackupDeleteRequest,
+) (*api.SdkCloudBackupDeleteResponse, error) {
+
+	if len(req.GetBackupId()) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Must provide backup id")
+	} else if len(req.GetCredentialUuid()) == 0 {
```
Comment (ipochi) [inline api/server/sdk/cloud_backup.go]: Correct me if i am wrong
pxctl doesnt place a mandatory requirement on providing cred uuid if there is only one credential present in the cluster.

Shouldn't it be the same for SDK ?

File: api/server/sdk/cloud_backup.go [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+		Contents: r.Contents,
+	}, nil
+}
+
+// History returns ??
+func (s *CloudBackupServer) History(
+	ctx context.Context,
+	req *api.SdkCloudBackupHistoryRequest,
+) (*api.SdkCloudBackupHistoryResponse, error) {
+
```
Comment (ipochi) [inline api/server/sdk/cloud_backup.go]: check if volume id is provided or not ?

File: api/server/sdk/cloud_backup.go [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+	return r.ToSdkCloudBackupEnumerateResponse(), nil
+}
+
+// Status provides status on a backup
+func (s *CloudBackupServer) Status(
+	ctx context.Context,
+	req *api.SdkCloudBackupStatusRequest,
+) (*api.SdkCloudBackupStatusResponse, error) {
+
+	r, err := s.driver.CloudBackupStatus(&api.CloudBackupStatusRequest{
```
Comment (ipochi) [inline api/server/sdk/cloud_backup.go]: check for request params local and volumeId

File: api/server/sdk/cloud_backup.go (hunk 4 above)
Comment (ipochi) [reply api/server/sdk/cloud_backup.go]: on second thoughts , I might disagree with my own statement.
but food for thought.

File: api/api.proto (hunk 1 above)
Comment (lpabon) [reply api/api.proto]: No, my mistake. I'll fix that.

File: api/server/sdk/cloud_backup.go (hunk 2 above)
Comment (lpabon) [reply api/server/sdk/cloud_backup.go]: yeah, good point.

File: api/server/sdk/cloud_backup.go (hunk 3 above)
Comment (lpabon) [reply api/server/sdk/cloud_backup.go]: Yeah, I understand what you mean, but it is ok.

File: api/server/sdk/cloud_backup.go (hunk 5 above)
Comment (lpabon) [reply api/server/sdk/cloud_backup.go]: yeah, I missed that. Thanks.

File: api/server/sdk/cloud_backup.go (hunk 6 above)
Comment (lpabon) [reply api/server/sdk/cloud_backup.go]: The object comments say that the srcvolumeid is optional. But those may be incorrect. I will check with @talakad .

Comment (talakad) [reply api/server/sdk/cloud_backup.go]: srcvolumeid is optional. if not mentioned, it will return status of all instead of requested src volumeid.

**Pull request merged**